### DISCUSSION TITLE
[@types/react-dom]/server renderToString takes incorrect type

### DISCUSSION THREAD

**Discussion by iamogbz** [DefinitelyTyped/DefinitelyTyped#57358] [Issues with a @types package]
`react-dom/server` `renderToString` method supports objects of the type `React.ReactNode` and we have been using it as such;
[renderToString](https://github.com/facebook/react/blob/c1220ebdde506de91c8b9693b5cb67ac710c8c89/packages/react-dom/src/server/ReactDOMStringRenderer.js#L11-L24) → [ReactDOMServerRenderer.constructor](https://github.com/facebook/react/blob/c1220ebdde506de91c8b9693b5cb67ac710c8c89/packages/react-dom/src/server/ReactPartialRenderer.js#L783-L784) → [flattenTopLevelChildren](https://github.com/facebook/react/blob/c1220ebdde506de91c8b9693b5cb67ac710c8c89/packages/react-dom/src/server/ReactPartialRenderer.js#L304-L307) → [ReactDOMServerRenderer.read](https://github.com/facebook/react/blob/c1220ebdde506de91c8b9693b5cb67ac710c8c89/packages/react-dom/src/server/ReactPartialRenderer.js#L964) → [ReactDOMServerRenderer.render](https://github.com/facebook/react/blob/c1220ebdde506de91c8b9693b5cb67ac710c8c89/packages/react-dom/src/server/ReactPartialRenderer.js#L1003-L1007)

However after installing the DT library we get typescript errors due to the type specified being the subtype `React.ReactElement`;
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/459bc4a2904d8c8c864976543d987fec3d24e159/types/react-dom/server.d.ts#L21

It's a bit counter to the [`renderToString` documentation](https://reactjs.org/docs/react-dom-server.html#rendertostring) which reads as;
> Render a ReactElement to its initial HTML. This should only be used on the server.

Not sure if reaching out to React team to clarify their documentation would be the solution, at least fixing the types in DT will resolve it for devs that use both libraries.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
For React 18 we can widen it, since they also use ReactNode for their own types (https://github.com/facebook/react/blob/c1220ebdde506de91c8b9693b5cb67ac710c8c89/packages/react-dom/src/server/ReactDOMLegacyServerBrowser.js#L93).

For React 17 we should stick to the documentation.

**Reply by iamogbz** [reply]
The documentation is "wrong" though, although I solved it in my use-case by overriding the method definition. It's just surprising to have a legitimate use-case start failing due to TS errors when bumping versions.
